linux/LinuxProcessList.c

```diff
@@ -1824,8 +1824,33 @@ static inline double LinuxProcessList_scanCPUTime(ProcessList* super) {
    return period;
 }
 
+static bool cpuIsOnline(unsigned int cpuId) {
+      /* if there's no sysfs file for the CPU node indicating online/offline status,
+       * the CPU is probably online, so assume online by default
+       */
+      bool online = 1;
+      char pathBuffer[64];
+
+      xSnprintf(pathBuffer, sizeof(pathBuffer), "/sys/devices/system/cpu/cpu%u/online", cpuId);
+
+      FILE* file = fopen(pathBuffer, "r");
+      unsigned int readval;
+      if (file) {
+         if (fscanf(file, "%u", &readval) == 1) {
+            online = readval;
+         }
+         fclose(file);
+      }
+
+      return online;
+}
+
+static int getConfiguredCPUCount(void) {
+   return (int) sysconf(_SC_NPROCESSORS_CONF);
+}
```

Comment:
This should be available globally already IIRC.

Reply:
I tried looking for that but couldn't find it.

ProcessList has [cpuCount](https://github.com/htop-dev/htop/blob/8154125d4bf26fff20241eebe17954254409a522/ProcessList.h#L84), which is updated on Linux by [LinuxProcessList_updateCPUcount](https://github.com/htop-dev/htop/blob/8154125d4bf26fff20241eebe17954254409a522/linux/LinuxProcessList.c#L167), but that number only includes the number of active CPUs, and using it as the upper limit for the CPU id leads to precisely the problem that I'm trying to solve.

Any pointers to where this would be available globally?